addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/JSFunctionPassiveScannerUnitTest.java

```diff
@@ -0,0 +1,163 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import net.htmlparser.jericho.Source;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class JSFunctionPassiveScannerUnitTest extends PassiveScannerTest<JSFunctionPassiveScanner> {
+
+    @Override
+    protected JSFunctionPassiveScanner createScanner() {
+        return new JSFunctionPassiveScanner();
+    }
+
+    @Override
+    public void setUpZap() throws Exception {
+        super.setUpZap();
+
+        Path xmlDir =
+                Files.createDirectories(
+                        Paths.get(Constant.getZapHome(), JSFunctionPassiveScanner.FUNC_LIST_DIR));
+        Path testFile = xmlDir.resolve(JSFunctionPassiveScanner.FUNC_LIST_FILE);
+        Files.write(testFile, Arrays.asList("# Test File", "bypassSecurityTrustHtml", "eval"));
+    }
+
+    @Test
+    public void shouldAlertGivenFunctionInJavaScriptResponse()
+            throws HttpMalformedHeaderException, URIException {
+        // Given
+        String body = "Some text <script>$eval()</script>\nLine 2\n";
+        HttpMessage msg = createHttpMessageWithRespBody(body, "text/javascript;charset=ISO-8859-1");
+        Source source = createSource(msg);
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldAlertGivenFunctionInHtmlScriptElements()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String body =
+                "<h1>Some text <script>Some Html Element bypassSecurityTrustHtml()</script></h1>\n"
+                        + "<b>No script here</b>\n";
+        HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+
+        assertTrue(msg.getResponseHeader().isText());
```

Comment:
I'd not add these, the test would fail anyway.

Reply:
I don't quite get what you mean by this, are you referring to the test case or the specific assert method in this line?

Reply:
These (this line and following) specific assertions.